File: Samples/CS/CodeSamples/SendSMS.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Inetlab.SMPP;
using Inetlab.SMPP.Common;
using Inetlab.SMPP.Headers;
using Inetlab.SMPP.Parameters;
using Inetlab.SMPP.PDU;

namespace CodeExamples
{
    public static class SendSmsExamples
    {
        public static async Task SendText(SmppClient client)
        {
          var resp = await client.Submit(
                SMS.ForSubmit()
                    .From("short_code")
                    .To("436641234567")
                    .Coding(DataCodings.UCS2)
                    .Text("test text")
                );

        }

        public static async Task SendBinary(SmppClient client)
        {
            byte[] data = ByteArray.FromHexString(
                "FFFF002830006609EC592F55DCE9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000005C67");

            var resp = await client.Submit(
                SMS.ForSubmit()
                    .From("short_code")
                    .To("436641234567")
                    .Data(data)
            );
        }

        public static async Task SendMessageToApplicationPort(SmppClient client)
        {

            var resp = await client.Submit(
                SMS.ForSubmit()
                    .From("short_code")
                    .To("436641234567")
                    .Text("test")
                    .Set(sm=>sm.UserData.Headers.Add(new ApplicationPortAddressingScheme16bit(0x1579, 0x0000) ))
            );
        }

        public static void AddVendorSpecificParameter(SubmitSm sm)
        {
            //0x1400 - 0x3FFF Reserved for SMSC Vendor specific optional parameters
            byte[] value = new byte[]{ 0x01};

            sm.Parameters.Add(new TLV(0x1410, value));
        }


        public static async Task SendSubmitSmWithUDH(SmppClient client)
        {
            //If you have UserDataHeader as byte array. You can create SubmitSm manually and pass it to headers collection.

            byte[] udh = new byte[]
            {
                5, //UDHL,
                0, //Concatenation
                3, // Length
                50, //message reference
                1, //total parts,
                1, // part number
            };



            SubmitSm sm = new SubmitSm();
            sm.SourceAddress = new SmeAddress("My Service");
            sm.DestinationAddress = new SmeAddress("+7917123456");
            sm.DataCoding = DataCodings.UCS2;
            sm.RegisteredDelivery = 1;

            sm.UserData.ShortMessage = client.EncodingMapper.GetMessageBytes("test message", sm.DataCoding);
            sm.UserData.Headers = udh;

            var resp = await client.Submit(sm);
        }

    }
}

File: Samples/VB.NET/CodeSamples/SendSMS.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Threading.Tasks
Imports Inetlab.SMPP
Imports Inetlab.SMPP.Common
Imports Inetlab.SMPP.Headers
Imports Inetlab.SMPP.Parameters
Imports Inetlab.SMPP.PDU

Namespace CodeExamples
	Public Module SendSmsExamples
		Public Async Function SendText(ByVal client As SmppClient) As Task
		  Dim resp = Await client.Submit(SMS.ForSubmit().From("short_code").To("436641234567").Coding(DataCodings.UCS2).Text("test text"))

		End Function

		Public Async Function SendBinary(ByVal client As SmppClient) As Task
			Dim data() As Byte = ByteArray.FromHexString("FFFF002830006609EC592F55DCE9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000005C67")

			Dim resp = Await client.Submit(SMS.ForSubmit().From("short_code").To("436641234567").Data(data))
		End Function

		Public Async Function SendMessageToApplicationPort(ByVal client As SmppClient) As Task

			Dim resp = Await client.Submit(SMS.ForSubmit().From("short_code").To("436641234567").Text("test").Set(Sub(sm) sm.UserData.Headers.Add(New ApplicationPortAddressingScheme16bit(&H1579, &H0))))
		End Function

		Public Sub AddVendorSpecificParameter(ByVal sm As SubmitSm)
			'0x1400 - 0x3FFF Reserved for SMSC Vendor specific optional parameters
			Dim value() As Byte = { &H1}

			sm.Parameters.Add(New TLV(&H1410, value))
		End Sub


		Public Async Function SendSubmitSmWithUDH(ByVal client As SmppClient) As Task
			'If you have UserDataHeader as byte array. You can create SubmitSm manually and pass it to headers collection.

			Dim udh() As Byte = { 5, 0, 3, 50, 1, 1}



			Dim sm As New SubmitSm()
			sm.SourceAddress = New SmeAddress("My Service")
			sm.DestinationAddress = New SmeAddress("+7917123456")
			sm.DataCoding = DataCodings.UCS2
			sm.RegisteredDelivery = 1

			sm.UserData.ShortMessage = client.EncodingMapper.GetMessageBytes("test message", sm.DataCoding)
			sm.UserData.Headers = udh

			Dim resp = Await client.Submit(sm)
		End Function

	End Module
End Namespace
